jaeger-core/src/test/java/com/uber/jaeger/ConfigurationTest.java

```diff
@@ -320,6 +339,490 @@ public void testPropagationValidFormat() {
     assertNotNull(textMap.get("uber-trace-id"));
   }
 
+  @Test
+  public void testEnsureCachedTracerInstance() {
+    Configuration configuration = new Configuration("the-service");
+    io.opentracing.Tracer tracer = configuration.getTracer();
+    assertEquals("Same instance of the tracer was expected", tracer, configuration.getTracer());
+  }
+
+  @Test
+  public void testEnsureNewTracerInstance() {
+    Configuration configuration = new Configuration("the-service");
+    io.opentracing.Tracer tracer = configuration.getTracerBuilder().build();
+    assertNotEquals("Different instance of the tracer was expected", tracer, configuration.getTracerBuilder().build());
+  }
+
+  @Test
+  public void testCloseWhenNoTracerWasBuilt() {
+    // this is a noop, but we ensure there's no NPE going on
+    new Configuration("the-service").closeTracer();
+  }
+
+  @Test
+  public void testClose() throws SenderException {
+    Sender sender = mock(Sender.class);
+    ReporterConfiguration reporterConfiguration = new ReporterConfiguration(sender);
```

Comment:
Should the constructor be removed (deprecated) - so only use builders?

Reply:
deprecated

Reply:
Added `@Deprecated` to this constructor